cloudahoywriter.lua

```diff
@@ -375,6 +409,42 @@ function maybe_write_data()
     lastWriteTime = os.clock()
 end
 
+-- Called from show_ui() on every draw. Keep this fast.
+-- Decides whether to automatically start or stop recording.
+function automatic_recording_state_check()
+    if CAWR_isPaused == 1 then return end
+
+    if not CAWR_groundSpeed then
+        return -- Possible race with registering datarefs
```

Comment:
wouldn't this potential race also happen with CAWR_isPaused?

Reply:
That was a change accidentally left in when I ran into a bug. The real bug was that I typo'd CAWR_groundspeed instead of CAWR_groundSpeed. I thought it was a race condition issue.

I don' think there's a race possible here. The `initialize_datarefs()` method is called before we register the `do_every_draw()` hook at the end of the file.